tools/actions_local_runner.py

```diff
@@ -70,65 +78,169 @@ def find_changed_files() -> List[str]:
     return [x.strip() for x in all_files if x.strip() != ""]
 
 
-async def run_step(step: Dict[str, Any], job_name: str, files: Optional[List[str]], quiet: bool) -> bool:
+def print_results(job_name: str, passed: bool, streams: List[str]) -> None:
+    header(job_name, passed)
+    for stream in streams:
+        stream = stream.strip()
+        if stream != "":
+            print(stream)
+
+
+async def shell_cmd(
+    cmd: str, env: Optional[Dict[str, Any]] = None, redirect: bool = True
+) -> Tuple[bool, str, str]:
+    proc = await asyncio.create_subprocess_shell(
+        cmd,
+        shell=True,
+        cwd=REPO_ROOT,
+        env=env,
+        stdout=subprocess.PIPE if redirect else None,
+        stderr=subprocess.PIPE if redirect else None,
+        executable=shutil.which("bash"),
+    )
+    stdout, stderr = await proc.communicate()
+
+    passed = proc.returncode == 0
+    if not redirect:
+        return passed, "", ""
+
+    return passed, stdout.decode(), stderr.decode()
+
+
+def header(name: str, passed: bool) -> None:
+    PASS = color(col.GREEN, "\N{check mark}")
+    FAIL = color(col.RED, "x")
+    icon = PASS if passed else FAIL
+    print(f"{icon} {color(col.BLUE, name)}")
+
+
+def get_flake_excludes() -> List[str]:
+    config = configparser.ConfigParser()
+    config.read(os.path.join(REPO_ROOT, ".flake8"))
+
+    excludes = config["flake8"]["exclude"].split("\n")
+    return [i.strip() for i in excludes if i.strip() != ""]
+
+
+async def run_flake8(files: Optional[List[str]], quiet: bool) -> bool:
+    cmd = "flake8"
+    if files is not None:
+        excludes = get_flake_excludes()
+        files = [
+            f
+            for f in files
+            if not any(is_match(pattern=exclude, filename=f) for exclude in excludes)
```

Comment:
at first glance, I don't think `is_match` is the correct function to use here, for a couple reasons:

1. as #55702 says, its implementation specifically calls [`mypy.config_parse.split_and_match_files`](https://github.com/python/mypy/blob/v0.770/mypy/config_parser.py#L45-L63) to "expand" wildcards into lists of paths (or just return the original widcard
   ```py
   >>> mypy.config_parser.split_and_match_files('*.pyi')
   ['*.pyi']
   ```
2. after expanding wildcards, it basically just checks if the 
   ```py
   >>> tools.mypy_wrapper.is_match(pattern='*.pyi', filename='torch/fx/__init__.pyi')                                                                   
   False
   ```
3. this one may not apply (it depends on the nature of the `files` passed to this function), but note that even without wildcards, the `is_match` function assumes that paths are relative to the repo root:
   ```py
   >>> tools.mypy_wrapper.is_match(pattern='./caffe2', 
   filename=str(Path('caffe2/python/models/download.py').resolve()))
   False
   ```

I think the key to all of this is that `mypy`'s list of `files` just acts differently from Flake8's `exclude`s (see also #55178 for more context), so you probably want to use a different function here.

Reply:
After doing a bunch of digging, flake8 uses [`fnmatch`](https://docs.python.org/3/library/fnmatch.html) to filter out files, but even that still doesn't work for files without globs: `fnmatch.fnmatch(name="./caffe2/test.py", pat="./caffe2")` so there's something I'm still missing. Either way, flake8 seems to still use excludes when explicitly passed a file:

1. add some trailing spaces to `torch/_C/_autograd.pyi`
2. `flake8 torch/_C/_autograd.pyi` -> no errors
3. remove `*.pyi` from `.flake8`
4. `flake8 torch/_C/_autograd.pyi` -> lots of errors

So all of this logic is useless for flake8